lib/ex_unit/lib/ex_unit/formatter.ex

```diff
@@ -138,20 +139,18 @@ defmodule ExUnit.Formatter do
     format_assertion_error(%{}, struct, [], :infinity, fn _, msg -> msg end, "")
   end
 
-  def format_assertion_error(test, struct, stack, width, formatter, counter_padding) do
+  defp format_assertion_error(test, struct, stack, width, formatter, counter_padding) do
     label_padding_size = if has_value?(struct.right), do: 7, else: 6
     padding_size = label_padding_size + byte_size(@counter_padding)
     inspect = &inspect_multiline(&1, padding_size, width)
-    {left, right} = format_sides(struct, formatter, inspect, padding_size, width)
 
     [
-      note: if_value(struct.message, &format_message(&1, formatter)),
-      doctest: if_value(struct.doctest, &code_multiline(&1, 2 + byte_size(@counter_padding))),
-      code: if_value(struct.expr, &code_multiline(&1, padding_size)),
-      code: unless_value(struct.expr, fn -> get_code(test, stack) || @no_value end),
-      arguments: if_value(struct.args, &format_args(&1, width)),
-      left: left,
-      right: right
+      {:note, if_value(struct.message, &format_message(&1, formatter))},
+      {:doctest, if_value(struct.doctest, &code_multiline(&1, 2 + byte_size(@counter_padding)))},
+      {:code, if_value(struct.expr, &code_multiline(&1, padding_size))},
+      {:code, unless_value(struct.expr, fn -> get_code(test, stack) || @no_value end)},
+      {:arguments, if_value(struct.args, &format_args(&1, width))}
+      | format_context(struct, formatter, inspect, padding_size, width)
```

Comment:
You can keep the same syntax as before and do right after:

```
[
  ...
]
|> Kernel.++(format_context(struct, formatter, inspect, padding_size, width))
|> ...
```

The behaviour and performance would be the same. :)